src/thor/trace_route_action.cc

```diff
@@ -392,16 +392,19 @@ void thor_worker_t::build_route(
       const auto* prev_segment = i > 0 ? path.second[i - 1] : nullptr;
       const auto* segment = path.second[i];
       const auto* next_segment = i < path.second.size() - 1 ? path.second[i + 1] : nullptr;
-      // if we uturn onto this edge we must trim the beginning and if we uturn off we trim the end
-      bool uturn_onto =
-          prev_segment && prev_segment->edgeid != segment->edgeid && prev_segment->target < 1.f;
-      bool uturn_off_of =
-          next_segment && segment->edgeid != next_segment->edgeid && segment->target < 1.f;
-      if (uturn_onto || uturn_off_of) {
-        edge_trimming[i] = {{uturn_onto, match_results[segment->first_match_idx].lnglat,
```

Comment:
as @dnesbitt61 mentioned the match index can be `-1` but only in the case that we dont need to trim. sadly the trimming datastructure is a pair and so you have to specify both first and second. if though one of them doesnt need to be set in this case. to get around this i assign one at a time and rely on the default initializer to intialize the pair with false and invalid lat lon etc